sources/osg/AutoTransform.js

```diff
@@ -0,0 +1,310 @@
+        this._matrixDirty = true;
+        this.dirtyBound();
+    },
+
+    getRotation: function () {
+        return this._rotation;
+    },
+
+    setScale: function ( scale ) {
+        this.setScaleFromVec3( Vec3.createAndSet( scale, scale, scale ) );
+    },
+
+    setScaleFromVec3: function ( scaleVec ) {
+        this._scale = scaleVec;
+        this._matrixDirty = true;
+        this.dirtyBound();
+    },
+
+    getScale: function () {
+        return this._scale;
+    },
+
+    setMinimumScale: function ( minimumScale ) {
+        this._minimumScale = minimumScale;
+    },
+
+    getMinimumScale: function () {
+        return this._minimumScale;
+    },
+
+    setMaximumScale: function ( maximumScale ) {
+        this._maximumScale = maximumScale;
+    },
+
+    getMaximumScale: function () {
+        return this._maximumScale;
+    },
+
+    setAutoScaleToScreen: function ( autoScaleToScreen ) {
+        this._autoScaleToScreen = autoScaleToScreen;
+        this._matrixDirty = true;
+    },
+
+    getAutoScaleToScreen: function () {
+        return this._autoScaleToScreen;
+    },
+
+    setAutoRotateToScreen: function ( value ) {
+        this._autoRotateToScreen = value;
+    },
+
+    getAutoRotateToScreen: function () {
+        return this._autoRotateToScreen;
+    },
+
+    setAutoScaleTransitionWidthRatio: function ( autoScaleTransitionWidthRatio ) {
+        this._autoScaleTransitionWidthRatio = autoScaleTransitionWidthRatio;
+    },
+
+    getAutoScaleTransitionWidthRatio: function () {
+        return this._autoScaleTransitionWidthRatio;
+    },
+
+    // local to "local world" (not Global World)
+    computeLocalToWorldMatrix: function ( matrix /*, nodeVisitor */ ) {
+        if ( this._matrixDirty ) this.computeMatrix();
+        if ( this.referenceFrame === TransformEnums.RELATIVE_RF ) {
+            Matrix.preMult( matrix, this._matrix );
+        } else {
+            Matrix.copy( this._matrix, matrix );
+        }
+    },
+
+    computeMatrix: ( function () {
+        var neg = Vec3.create();
+        return function () {
+            if ( !this._matrixDirty ) return;
+            Matrix.makeRotateFromQuat( this._rotation, this._matrix );
+            //_cachedMatrix.postMultTranslate(_position);
+            Matrix.postMultTranslate( this._matrix, this._position );
+            Matrix.preMultScale( this._matrix, this._scale );
+            Matrix.preMultTranslate( this._matrix, Vec3.neg( this._pivotPoint, neg ) );
+            this._matrixDirty = false;
+        };
+
+    } )(),
+
+    computeWorldToLocalMatrix: ( function () {
+        var neg = Vec3.create();
+        var rotInverse = Quat.create();
+        var scaleInverse = Vec3.create();
+        return function ( matrix /*, nodeVisitor */ ) {
+            if ( this.scale[ 0 ] === 0.0 && this.scale[ 1 ] === 0.0 && this.scale[ 2 ] === 0.0 ) {
+                return false;
+            }
+            scaleInverse[ 0 ] = 1 / this._scale[ 0 ];
```

Comment:
use 1.0 instead of 1 optimization to avoid the vm use int and cast to float after